v2/pkg/templates/compile.go

```diff
@@ -35,44 +38,25 @@ func Parse(filePath string, preprocessor Preprocessor, options protocols.Execute
 		return value.(*Template), err
 	}
 
-	template := &Template{}
-
-	data, err := utils.ReadFromPathOrURL(filePath, options.Catalog)
-	if err != nil {
-		return nil, err
-	}
-
-	data = template.expandPreprocessors(data)
-	if preprocessor != nil {
-		data = preprocessor.Process(data)
+	var reader io.ReadCloser
+	if utils.IsURL(filePath) {
+		resp, err := http.Get(filePath)
+		if err != nil {
+			return nil, err
+		}
+		reader = resp.Body
+	} else {
+		var err error
+		reader, err = options.Catalog.OpenFile(filePath)
+		if err != nil {
+			return nil, err
+		}
 	}
-
-	if err := yaml.Unmarshal(data, template); err != nil {
+	defer reader.Close()
+	template, err := ParseFromReader(reader, preprocessor, options)
+	if err != nil {
 		return nil, err
 	}
-
-	if utils.IsBlank(template.Info.Name) {
```

Comment:
Shouldn't also this logic be in the new `ParseFromReader` since it performs additional checks and fills other fields that would be missed otherwise?

Reply:
yes.. it is already there in `ParseFromReader` function.